orchagent/portsorch.cpp

```diff
@@ -5072,6 +5077,46 @@ void PortsOrch::generatePriorityGroupMapPerPort(const Port& port)
     CounterCheckOrch::getInstance().addPort(port);
 }
 
+void PortsOrch::generatePortCounterMap()
+{
+    if (m_isPortCounterMapGenerated)
+    {
+        return;
+    }
+
+    auto port_counter_stats = generateCounterStats(PORT_STAT_COUNTER_FLEX_COUNTER_GROUP);
+    for (const auto& it: m_portList)
+    {
+        if (it.second.m_type != Port::Type::PHY)
```

Comment:
Please comment why only PHY port needs to be queried.

Reply:
Generating counter stats will eventually trigger syncd to query the counter statistics from the HW and this is possible only for physical ports.

Reply:
Please add comment in code so that people will understand this logic in the future.

Reply:
Added.